### ISSUE TITLE
Parallel Multi Instance task graph with spurious GW

### ISSUE THREAD

**Issue by asauve2** [sartography/SpiffWorkflow#331]
I'm looking forward for using threading with spiffworkflow-1.2.1.

In the process of understanding how PMI work I printed the task graph of a simple 
workflow which was
```
(start) -> [ Activity_PMI ||| ] -> (End)
```

BPMN edited with camunda 7 (I'm aware it's not 100% supported) and set loop count to 3.

The tree is plotted using `workflow.get_tasks_iterator()` and next `task.children`.
Task identity was checked using the python instance address as duplicates appeared.
The result graph looks like that:
```
    Root
    |
    Start
    |
    StartEvent
    |
    GW_Start ------------------------------             # Up to here everything looks fine
    |                   |                 |
    Activity_PMI     Activity_PMI_0      Activity_PMI_1 # Indexation here start to look fuzzy
    |                   |                 |
    GW_end              GW_end            GW_end        # 3 GW_end is this normal?
    |                   |                 |
    EndEvent            EndEvent----------              # the 2nd end event connects only 2 tasks
                        |                               # Shouldn't it be only one EndEvent ?
                        EndJoin
                        |
                        End

    EndJoin                                             # A spurious disconnected EndJoin appears?
    |
    End
```

In practice execution with `workflow.do_engine_steps()` triggers the correct
number of tasks and inputs seems correct sequentially.
But the task graph has clearly an issue.

**Comment by danfunk** [comment]
Great talking to you today @asauve2 .
We have, as of this morning, published a pre-release [2.0.0rc0](https://pypi.org/project/SpiffWorkflow/2.0.0rc0/) which corrects this issue.  [Release notes](https://github.com/sartography/SpiffWorkflow/releases/tag/v2.0.0rc0) and updated [documentation](https://spiffworkflow.readthedocs.io/en/v2.0.0rc0/bpmn/overview.html) are also on line.

**Comment by asauve2** [comment]
That was great for us too @danfunk.
This specific issue is not yet tested, but when trying out [2.0.0rc0](https://github.com/sartography/SpiffWorkflow/tree/v2.0.0rc0) and checking current
documentation on how to implement custom spec classes I realized that we have now
to extend from `SpiffWorkflow.spiff.specs.spiff_task.SpiffBpmnTask`
It solves several other top level issues like the correct `__init__()` call and the building
of intermediate boundary error task sub graph.

Keep you informed when reaching PMI stuff.

**Comment by asauve2** [comment]
We have now transitionned our code and passed again our 120 tests with spiff-v2.0.0rc0.
Most of it was effortless except for boundary error events requiring special idiomatic form 
as answered in #334 .

The new PMI code builds now a clean task graph so this issue can be considered closed in v2.0.0rc0.
However when iterating over tasks using loopCardinality BPMN attribute, the loop index
does not seem to be accessible from Task data. This is discussed in separate issue #337 for clarity.
